src/test/java/com/thebois/models/world/generation/noises/PerlinNoiseTests.java

```diff
@@ -0,0 +1,217 @@
+    public void perlinNoiseSameResultWithSameValues(
+        final float posX, final float posY) {
+        // Arrange
+        final PerlinNoise perlinNoise = new PerlinNoise(1, 1, 1, 1);
+        final float heightValue1;
+        final float heightValue2;
+
+        // Act
+        heightValue1 = perlinNoise.sample(posX, posY);
+        heightValue2 = perlinNoise.sample(posX, posY);
+
+        // Assert
+        assertThat(heightValue1).isEqualTo(heightValue2);
+    }
+
+    @ParameterizedTest
+    @MethodSource("getPerlinNoiseSettingsParameters")
+    public void perlinNoiseSameResultWithSameValues(
+        final int octave, final float amplitude, final float frequency, final float persistence) {
+        // Arrange
+        final PerlinNoise perlinNoise = new PerlinNoise(octave, amplitude, frequency, persistence);
+        final int x = 1;
+        final int y = 1;
+        final float heightValue1;
+        final float heightValue2;
+
+        // Act
+        heightValue1 = perlinNoise.sample(x, y);
+        heightValue2 = perlinNoise.sample(x, y);
+
+        // Assert
+        assertThat(heightValue1).isEqualTo(heightValue2);
+    }
+
+    @ParameterizedTest
+    @MethodSource("getTwoDifferentPositions")
+    public void perlinNoiseDifferentResultWithDifferentValues(
+        final float posX1, final float posY1, final float posX2, final float posY2) {
+        // Arrange
+        final PerlinNoise perlinNoise = new PerlinNoise(1, 1, 1, 1);
+        final float heightValue1;
+        final float heightValue2;
+
+        // Act
+        heightValue1 = perlinNoise.sample(posX1, posY1);
+        heightValue2 = perlinNoise.sample(posX2, posY2);
+
+        // Assert
+        assertThat(heightValue1).isNotEqualTo(heightValue2);
+    }
+
+    @ParameterizedTest
+    @MethodSource("getPositions")
+    public void perlinNoiseDifferentResultWithSameValuesAndDifferentSeed(
+        final float x, final float y) {
+        // Arrange
+        final PerlinNoise perlinNoise = new PerlinNoise(1, 1, 1, 1);
+        final int seed1 = 0;
+        final int seed2 = 1;
+        perlinNoise.setSeed(seed1);
+        final float heightValue1;
+        final float heightValue2;
+
+        // Act
+        heightValue1 = perlinNoise.sample(x, y);
+        perlinNoise.setSeed(seed2);
+        heightValue2 = perlinNoise.sample(x, y);
+
+        // Assert
+        assertThat(heightValue1).isNotEqualTo(heightValue2);
+    }
+
+    @ParameterizedTest
+    @MethodSource("getPositions")
+    public void perlinNoiseDifferentResultWithSameValuesAndDifferentSettings(
+        final float x, final float y) {
+        // Arrange
+        final PerlinNoise perlinNoise1 = new PerlinNoise(1, 1, 1, 1);
+        final PerlinNoise perlinNoise2 = new PerlinNoise(2, 2, 2, 2);
+        final float heightValue1;
+        final float heightValue2;
+
+        // Act
+        heightValue1 = perlinNoise1.sample(x, y);
+        heightValue2 = perlinNoise2.sample(x, y);
```

Comment:
style

I don't see the point in splitting up the declaration from the definition? Feels like there are two extra lines of code for no reason.

Reply:
Same goes for the other tests.

Reply:
I have no clue why I'm doing it, but I started doing it and then I just tried to be consistent. Will change it and try to get rid of the habbit.